spring-cloud-vault-config/src/main/java/org/springframework/cloud/vault/config/VaultConfigDataLocationResolver.java

```diff
@@ -116,7 +119,25 @@ public boolean isResolvable(ConfigDataLocationResolverContext context, ConfigDat
 			contextPath = contextPath.substring(1);
 		}
 
-		return Collections.singletonList(new VaultConfigLocation(contextPath, location.isOptional()));
+		String[] splitContextPath = StringUtils.split(contextPath, "?prefix=");
```

Comment:
We should rather try using a Spring utility that properly parses query strings instead of having a duplicated approach to query string parsing. Have a look at `UriComponents` via `UriComponentsBuilder`.

Reply:
Reusing the UriComponentsBuilder now.